src/Zir.zig

```diff
@@ -248,6 +248,9 @@ pub const Inst = struct {
         /// Given a pointer type, returns its element type.
         /// Uses the `un_node` field.
         elem_type,
+        /// Given a vector type, returns its element type.
+        /// Uses the `un_node` field.
+        vector_elem_type,
```

Comment:
There is already elem_type above; why must there also be vector_elem_type?

Reply:
See https://github.com/ziglang/zig/pull/16569

Reply:
Since this is for RLS, it's important for the instruction to be quite precise for good error messages. This isn't one of those instructions where the type is already validated: this comes before the splat, so is doing the validation. For instance, if you try to do `@as(*@Vector(2, u32), @splat(123))`, you want the error to be that a vector type was expected, but if we use a general `elem_type`, we'd instead get `expected type '@Vector(2, u32)', found 'comptime_int'`. (Note that the error could seem even weirder if vectors aren't involved at all!)